### DISCUSSION TITLE
What files do I need to use in the yolov5 repository for my already existing code base

### DISCUSSION THREAD

**Discussion by GouthamChandrappa** [ultralytics/ultralytics#11857] [Q&A]
I have a code base for my Semi supervised federated machine learning project, 
Currently it is used only for image classification tasks, I need to integrate object detection as well.
I have cloned the repo of yolov5, and I have the Pascal VOC dataset, 
What Particular files of Yolov5 do I need to use inorder train the model on the dataset that is present in my codebase
I already have cloned Yolo.py and dataloader.py files into my codebase.

What files do I exactly need to use from the entire repo in order to complete this
I am a begginer, please do help me in this.
Thanks in advance

**Comment by glenn-jocher** [comment]
Hello! To integrate object detection using YOLOv5 into your existing codebase for the Pascal VOC dataset, you'll need a few more components from the YOLOv5 repository along with `Yolo.py` and `dataloader.py`:

1. **Configuration files**: Locate the YAML file in `yolov5/data/` that corresponds to your dataset or create one specifically for Pascal VOC. This file is crucial because it dictates the model's settings, including paths to your dataset.
2. **Model Configuration**: Choose a model configuration YAML file from `yolov5/models/` that suits your needs for detection, such as `yolov5s.yaml` for a smaller model.
3. **Training Script**: Use `train.py` from the root of the repository to train your model.
4. **Utils**: Make sure the `yolov5/utils/` directory is included as it contains essential modules and functions used across the project.

Here's a brief Python code snippet to help guide you on how to initiate the training:

```python
from yolov5 import train

# Assuming you have set up your dataset config and model config yaml files
train.run(data='path/to/your/voc.yaml', cfg='path/to/your/model.yaml', epochs=50)
```

If you're just starting out with YOLOv5, I highly recommend reviewing the [documentation](https://docs.ultralytics.com/) and exploring their comprehensive tutorials to better understand the workflow and capabilities. Good luck with your project! If you need more assistance or have other questions as you go along, feel free to ask. 😊
